tokio-console/src/config.rs

```diff
@@ -107,20 +112,66 @@ pub struct ViewOptions {
 }
 
 /// Toggles on and off color coding for individual UI elements.
-#[derive(Clap, Debug, Copy, Clone)]
+#[derive(Clap, Debug, Copy, Clone, Deserialize)]
 pub struct ColorToggles {
     /// Disable color-coding for duration units.
-    #[clap(long = "no-duration-colors", parse(from_flag = std::ops::Not::not), group = "colors")]
-    pub(crate) color_durations: bool,
+    #[clap(long = "no-duration-colors", group = "colors")]
+    #[serde(rename = "durations")]
+    color_durations: Option<bool>,
 
     /// Disable color-coding for terminated tasks.
-    #[clap(long = "no-terminated-colors", parse(from_flag = std::ops::Not::not), group = "colors")]
-    pub(crate) color_terminated: bool,
+    #[clap(long = "no-terminated-colors", group = "colors")]
+    #[serde(rename = "terminated")]
+    color_terminated: Option<bool>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct ConfigFile {
+    charset: Option<CharsetConfig>,
+    colors: Option<ColorsConfig>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct CharsetConfig {
+    lang: Option<String>,
+    ascii_only: Option<bool>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct ColorsConfig {
+    enabled: Option<bool>,
+    truecolor: Option<bool>,
+    palette: Option<Palette>,
+    enable: Option<ColorToggles>,
```

Comment:
nit: these are only used for parsing the TOML config file, they're not exposed to the rest of the crate, as we turn them into a `Config` before returning it. i think these don't need to be `pub`:
```suggestion
#[derive(Debug, Clone, Deserialize)]
struct ConfigFile {
    charset: Option<CharsetConfig>,
    colors: Option<ColorsConfig>,
}

#[derive(Debug, Clone, Deserialize)]
struct CharsetConfig {
    lang: Option<String>,
    ascii_only: Option<bool>,
}

#[derive(Debug, Clone, Deserialize)]
struct ColorsConfig {
    enabled: Option<bool>,
    truecolor: Option<bool>,
    palette: Option<Palette>,
    enable: Option<ColorToggles>,
```

Reply:
also, it might be worth adding comments explaining that these types are used so that the structure of the config file can be different from how the CLI arguments are structured?